platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,157 @@
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.PostConstruct;
+import lombok.AllArgsConstructor;
+import org.stellar.anchor.api.exception.ValueValidationException;
+import org.stellar.anchor.platform.data.PaymentObservingAccount;
+import org.stellar.anchor.util.Log;
+
+public class PaymentObservingAccountsManager {
+  Map<String, ObservingAccount> allAccounts;
+  private final PaymentObservingAccountStore store;
+
+  public PaymentObservingAccountsManager(PaymentObservingAccountStore store) {
+    this.store = store;
+    allAccounts = new ConcurrentHashMap<>();
+  }
+
+  @PostConstruct
+  public void initialize() throws ValueValidationException {
+    List<PaymentObservingAccount> accounts = store.list();
+    for (PaymentObservingAccount account : accounts) {
+      ObservingAccount oa =
+          new ObservingAccount(account.getAccount(), account.getLastObserved(), true);
+      upsert(oa);
+    }
+
+    // Start the eviction task
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    scheduler.scheduleAtFixedRate(
+        this::evictAndPersist, 60, getEvictPeriod().getSeconds(), TimeUnit.SECONDS);
+  }
+
+  public void evictAndPersist() {
+    Log.info("Evicting old accounts...");
+    this.evict(getEvictMaxAge());
+    Log.info("Persisting accounts...");
+    for (ObservingAccount account : this.getAccounts()) {
+      store.upsert(account.account, account.lastObserved);
+    }
+  }
+
+  /**
+   * Adds an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param account The account being observed.
+   * @param expiring true if the account can be expired. false, otherwise.
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.expiring.equals(existingAccount.expiring))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
```

Comment:
👀  Is this line needed?
```suggestion
```

Reply:
Probably not. But makes me worry if we do not have this line. 😓

Reply:
That's ok to leave it there then if we want to make things more explicit 👍